src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -239,41 +235,26 @@ private void flushData(final EventTypePartition pk, final List<ConsumedEvent> da
                            final Optional<String> metadata) {
         try {
             final NakadiCursor sentOffset = offsets.get(pk).getSentOffset();
-            final String batch = serializeBatch(sentOffset, data, metadata);
+            final SubscriptionCursor cursor = getContext().getCursorConverter().convert(
+                    sentOffset,
+                    getContext().getCursorTokenService().generateToken());
+
+            final int batchSize = getContext().getWriter().writeSubscriptionBatch(
+                    getOut().getOutputStream(),
+                    cursor,
+                    data,
+                    metadata);
+
+            getOut().getOutputStream().flush();
```

Comment:
I think this is the right place to be, since the writer is just a writer and should have no opinion on when something should be flushed or not.